azext_iot/operations/hub.py

```diff
@@ -246,6 +246,38 @@ def _create_self_signed_cert(subject, valid_days, output_path=None):
     return create_self_signed_certificate(subject, valid_days, output_path)
 
 
+def update_iot_device_custom(instance, edge_enabled=None, status=None, status_reason=None,
+                             auth_method=None, primary_thumbprint=None, secondary_thumbprint=None,
+                             primary_key=None, secondary_key=None):
+    if edge_enabled is not None:
+        instance['capabilities']['iotEdge'] = edge_enabled
+    if status is not None:
+        instance['status'] = status
+    if status_reason is not None:
+        instance['statusReason'] = status_reason
+    if auth_method is not None:
+        if auth_method == DeviceAuthType.shared_private_key.name:
+            auth = 'sas'
+            if (primary_key and not secondary_key) or (not primary_key and secondary_key):
```

Comment:
Does the service support providing your own SAS key for devices? From what I last recall, you are able to regenerate the primary or secondary key by setting it to an empty value. Let's validate whether you are able to provide a custom key here, and if not when the user changes from some other auth mode to shared private key, this functionality should automatically regenerate keys for them.

If custom shared private key is not possible, then we don't need the `primary_key` and `secondary_key` parameters. Instead we can have a `regenerate_keys` / `--regenerate-keys` where if `auth_method == DeviceAuthType.shared_private_key.name` this function will set both `instance['authentication']['symmetricKey']['primaryKey']=''` and `instance['authentication']['symmetricKey']['secondaryKey'] = ''` which _should_ regenerate them.

Reply:
Yes, we can provide custom keys

Reply:
OK. We should still support regenerating keys easily like in the Portal. 
![image](https://user-images.githubusercontent.com/28658112/92973849-ceea5b80-f439-11ea-8547-6475868a727e.png)

Reply:
Shouldn't we create a separate command for renewing the keys for the device-identity like https://docs.microsoft.com/en-us/cli/azure/iot/hub/policy?view=azure-cli-latest#az_iot_hub_policy_renew_key

Reply:
Yea thats fair.